apps/gitness/src/components/FileContentViewer.tsx

```diff
@@ -0,0 +1,167 @@
+import { useEffect, useMemo, useState } from 'react'
+import { CodeEditor } from '@harnessio/yaml-editor'
+import { themes } from '../pages/pipeline-edit/theme/monaco-theme'
+import copy from 'clipboard-copy'
+import { decodeGitContent, filenameToLanguage, formatBytes, getTrimmedSha } from '../utils/git-utils'
+import {
+  Button,
+  ButtonGroup,
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuTrigger,
+  Icon,
+  Spacer,
+  StackedList,
+  Text,
+  ToggleGroup,
+  ToggleGroupItem
+} from '@harnessio/canary'
+import { timeAgoFromISOTime } from '../pages/pipeline-edit/utils/time-utils'
+import { MarkdownViewer, PipelineStudioToolbarActions, TopDetails, TopTitle } from '@harnessio/playground'
+import { OpenapiGetContentOutput } from '@harnessio/code-service-client'
+
+interface FileContentViewerProps {
+  repoContent: OpenapiGetContentOutput
+}
+
+export type ViewTypeValue = 'preview' | 'code' | 'blame'
+
+const setDefaultView = (language: string): ViewTypeValue => {
+  return language === 'markdown' ? 'preview' : 'code'
+}
+
+export default function FileContentViewer({ repoContent }: FileContentViewerProps) {
+  const fileName = repoContent?.name || ''
+  const language = filenameToLanguage(fileName) || ''
+  const fileContent = decodeGitContent(repoContent?.content?.data)
+  const [view, setView] = useState<ViewTypeValue>(setDefaultView(language))
+  const [isEditMode, setIsEditMode] = useState<boolean>(false)
+  const latestFile = {
+    user: {
+      name: repoContent?.latest_commit?.author?.identity?.name || ''
+    },
+    lastCommitMessage: repoContent?.latest_commit?.message || '',
+    timestamp: repoContent?.latest_commit?.author?.when
+      ? timeAgoFromISOTime(repoContent?.latest_commit?.author.when)
+      : '',
+    sha: repoContent?.sha && getTrimmedSha(repoContent?.sha)
+  }
+
+  const themeConfig = useMemo(
+    () => ({
+      defaultTheme: 'dark',
+      themes
+    }),
+    []
+  )
+
+  useEffect(() => {
+    setView(setDefaultView(language))
```

Comment:
add error handling if language is missing or null?

Reply:
the getDefaultView takes care of that